### PR TITLE
Fix an issue with `capture_sample_data` and record type capturing.

### PR DESCRIPTION
**Pull request by prescod**
Record types were being extracted explicitly by the mapping file 
generator in addition to the magic record type handling. Now, the 
auto-extract does not treat record type ID fields as lookups but rather 
as plain text.

### CODE REVIEW

File: cumulusci/salesforce_api/filterable_objects.py [hunk 1]
```diff
@@ -34,6 +34,7 @@
     "PermissionSetLicenseAssign",
     "PermissionSetTabSetting",
     "Profile",
+    "RecordType",
```
Comment (prescod) [inline cumulusci/salesforce_api/filterable_objects.py:37]: Add to list of objects to NOT extract.

File: cumulusci/tasks/bulkdata/extract_dataset_utils/calculate_dependencies.py [hunk 2]
```diff
@@ -26,6 +26,7 @@ def _calculate_dependencies_for_declarations(
     """
     dependencies = {}
     for decl in decls:
+        assert isinstance(decl.sf_object, str)
```
Comment (prescod) [inline cumulusci/tasks/bulkdata/extract_dataset_utils/calculate_dependencies.py:29]: For type checker.

File: cumulusci/tasks/bulkdata/extract_dataset_utils/calculate_dependencies.py [hunk 3]
```diff
@@ -50,7 +51,7 @@ def _collect_dependencies_for_sobject(
         if not field_info.createable:  # pragma: no cover
             continue
         references = field_info.referenceTo
-        if len(references) == 1:
+        if len(references) == 1 and not references[0] == "RecordType":
```
Comment (prescod) [inline cumulusci/tasks/bulkdata/extract_dataset_utils/calculate_dependencies.py:54]: Never pull in a RecordType table.

File: cumulusci/tasks/bulkdata/generate_mapping_utils/generate_mapping_from_declarations.py [hunk 4]
```diff
@@ -33,13 +33,13 @@ def _mapping_step(decl):
         fields = tuple(chain(decl.fields, decl.lookups.keys()))
         return MappingStep(
             sf_object=decl.sf_object,
-            fields=zip(fields, fields),
+            fields=dict(zip(fields, fields)),
```
Comment (prescod) [inline cumulusci/tasks/bulkdata/generate_mapping_utils/generate_mapping_from_declarations.py:36]: Fix a type checker complaint.

File: cumulusci/tasks/bulkdata/generate_mapping_utils/generate_mapping_from_declarations.py [hunk 5]
```diff
@@ -33,13 +33,13 @@ def _mapping_step(decl):
...
             # lookups=lookups,      # lookups can be re-created later, for simplicity
         )
 
     mapping_steps = [_mapping_step(decl) for decl in simplified_decls_w_lookups]
 
-    mappings = generate_load_mapping_file(mapping_steps, intertable_dependencies, None)
+    mappings = generate_load_mapping_file(mapping_steps, intertable_dependencies, [])
```
Comment (prescod) [inline cumulusci/tasks/bulkdata/generate_mapping_utils/generate_mapping_from_declarations.py:42]: Type checker.

**Review by davidmreed — APPROVED** [review]

**Pull request merged**